### ISSUE TITLE
latest arrayvec versions fail build (>0.7.4): "type annotation needed"

### ISSUE THREAD

**Issue by notonamap** [paupino/rust-decimal#669]
should be an easy fix

https://github.com/paupino/rust-decimal/blob/dd711dbee5081a243e8eae5b314ecd8b0505f3ef/src/serde.rs#L254-L277

> Leveraging the existing `as_str` to implement `AsRef<Path>`, so that ArrayString can be used as a drop-in replacement for String in a few more places.

[#arrayvec](https://github.com/bluss/arrayvec/pull/218#issue-1206303206)

**Comment by shank03** [comment]
PR is opened for this fix.. till then remove `Cargo.lock` and try this:
```toml
arrayvec = "=0.7.4"
```

**Comment by shank03** [comment]
PR is opened, till then in case, 
- delete `target` folder
- and replace in `Cargo.lock`
```toml
[[package]]
name = "arrayvec"
version = "0.7.4"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "96d30a06541fbafbc7f82ed10c06164cfbd2c401138f6addd8404629c4b16711"
```

**Comment by discosultan** [comment]
When can we expect a crates.io release?

**Comment by paupino** [comment]
I've just released [1.36](https://crates.io/crates/rust_decimal/1.36.0) to crates.io so this should now be available downstream.

**Comment by discosultan** [comment]
Amazing, thank you
